protocol-units/da/m1/light-node/src/v1/sequencer.rs

```diff
@@ -200,20 +245,25 @@ impl LightNodeService for LightNodeV1 {
 		// make transactions from the blobs
 		let mut transactions = Vec::new();
 		for blob in blobs_for_submission {
-			let transaction : Transaction = serde_json::from_slice(&blob.data)
+			let transaction: Transaction = serde_json::from_slice(&blob.data)
 				.map_err(|e| tonic::Status::internal(e.to_string()))?;
 			transactions.push(transaction);
 		}
-		
-		// publish the transactions
-		for transaction in transactions {
-			debug!("Publishing transaction: {:?}", transaction.id());
 
-			self.memseq
-				.publish(transaction)
-				.await
-				.map_err(|e| tonic::Status::internal(e.to_string()))?;
-		}
+		// publish the transactions
+		let memseq = self.memseq.clone();
+		tokio::spawn(async move {
+			for transaction in transactions {
+				debug!("Publishing transaction: {:?}", transaction.id());
+				memseq
+					.publish(transaction)
+					.await
+					.map_err(|e| tonic::Status::internal(e.to_string()))?;
+			}
+			Ok::<(), tonic::Status>(())
+		}).await.map_err(
+			|e| tonic::Status::internal(e.to_string())
+		)??;
```

Comment:
This is a good example of how this way to define everything async can lead to inefficient code. This code block should be spawn_blocking because there's no async IO in the memseq publish process.
The` memseq.publish()` call `add_transaction` which came from the async trait `MempoolTransactionOperations`. And in the end it calls `RocksdbMempool::add_mempool_transaction()` which you `spawn_blocking` like for DB.
So instead of spawning one thread for all transaction to publish we spawn one thread for each transaction which can lead to worse performance than spawning nothing and with the current code it's very hard to change.
@mzabaluev, the new architecture should take care of that and async should be dissociated from sync function.
And I think it's for the same reason that you `spawn_blocking` each method of DB. I think they shouldn't be `spawn_blocking` because each method do a simple thing, and it's more the caller that could decide to `spawn_blocking` or not.
`spawn_blocking` too much short task can be less efficient than not doing it.

Reply:
Yes, but we were trying things. 

That being said, I haven't seen significant degradation with `spawn_blocking` calls, and after all they just add a closure to the block thread pool. 

For the sake of this PR, so that I don't have to remove all of the internal `spawn_blocking` calls, I can add a `publish_transactions` as I did for `pop_transactions`, so that the overhead of `spawn_blocking` is minimal.

Reply:
I did the above. I believe Tokio `spawn_blocking` is particularly good for blocking IO, which `rocksdb` has a lot of. For larger sets and gets like this, I'm an advocate.

Reply:
Yes `spawn_blocking` is very important for CPU bounder task, but it's use you be done with care because you can spawn a thousand of them very easily and performance can degrade very fast.
As you don't see perf decrease, it's ok for me. If was mostly to attract attention to this fact.